src/odemis/dataio/tiff.py

```diff
@@ -2613,3 +2627,127 @@ def _iterDirectories(tiff_file):
             tiff_file.ReadDirectory()
             dir_index += 1
             yield dir_index
+
+ 
+### ThermoFisher Metadata parsing
+def _parseINIMetadataAsDict(metadata: str, delimiter: str = "\r\n") -> dict:
+    """
+    Parses ini metadata as dictionary
+    :param metadata: metadata string as ini  
+    :param delimiter: delimiter between rows (optional, default: "\r\n")  
+    :return: metadata as dictionary
+    """
+    """
+    Metadata Format:
+        [subsystem]
+        k=v
+        ...
+    """
+    md = {}
+    for item in metadata.split(delimiter):
+        try:
+            if item == "":
+                continue # skip blank lines
+            elif re.match(r"\[(.*?)\]", item):
+                subsystem =  item[1:-1] # rm brackets
+                md[subsystem] = {}
+            else:
+                k, v = item.split("=")   
+                md[subsystem][k] = v
+        except Exception as e:
+            logging.debug("Failed to parse metadata item: %s", item)
+    return md
+
+
+def _convertTFStoOdemisMD(metadata: str) -> dict:
+    """
+    Converts ThermoFisher Scientific (TFS) metadata to odemis metadata format
+    :param: metadata : metadata string as ini (TFS format)
+    :return: metadata as dictionary (odemis format)
+    :raises KeyError: if the metadata cannot be parsed
+    """
+
+    # parse metadata as dictionary
+    fei_md = _parseINIMetadataAsDict(metadata)
```

Comment:
Could you try `configparser` again? (Did you see my comment about it?)

Reply:
resolved, was silly mistake on my end. now using configparser -> dict